main/EventSystem.cpp

```diff
@@ -2442,27 +2440,28 @@ void CEventSystem::ExportDomoticzDataToLua(lua_State *lua_state, uint64_t device
 			_hardwareNames[ID] = tlist;
 		}
 	}
+	
+	Json::Value tempjson;
+	m_webservers.GetJSonDevices(tempjson, "true", "", "dzvID", "", "", "", true, false, false, 0, "");
```

Comment:
Agreed, it's an expensive call. 
The problem is that we want the data fields which are only processed and created at the end of that call and put in the json. They are not available anywhere else (unless we duplicate the code).